packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -1450,4 +1450,29 @@ describe('TransactionController', () => {
       },
     );
   });
+
+  describe('initApprovals', () => {
+    it('creates approvals for all unapproved transaction', async () => {
+      const controller = newController();
+      await controller.addTransaction({
+        from: ACCOUNT_MOCK,
+        to: ACCOUNT_MOCK,
+      });
+
+      expect(controller.state.transactions[0].status).toBe(
+        TransactionStatus.unapproved,
+      );
+
+      controller.initApprovals();
+
+      expect(delayMessengerMock.call).toHaveBeenCalledTimes(2);
```

Comment:
Nit: to make this assertion more meaningful(not expecting the call two times), we could either reset call counts with `delayMessengerMock.mockClear()` after adding the first transaction or 
we could directly push the transaction to the state with 
```
controller.state.transactions.push({
        from: MOCK_PREFERENCES.state.selectedAddress,
        id: 'foo',
        networkID: '5',
        status: TransactionStatus.unapproved,
        transactionHash: '1337',
      } as any);
```
and expect that call to happen once.

Reply:
Ow thanks for that! I just included your suggestion in the test.